geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteFunctionOp.java

```diff
@@ -107,11 +106,8 @@ public static void execute(final PoolImpl pool,
 
         } catch (ServerConnectivityException se) {
 
-          if (maxRetryAttempts == PoolFactory.DEFAULT_RETRY_ATTEMPTS) {
-            // If the retryAttempt is set to default(-1). Try it on all servers once.
-            // Calculating number of servers when function is re-executed as it involves
-            // messaging locator.
-            maxRetryAttempts = pool.getConnectionSource().getAllServers().size() - 1;
+          if (maxRetryAttempts == -1) {
+            maxRetryAttempts = pool.calculateRetryAttempts(se);
```

Comment:
I didn't notice this before, but if we continue to get ServerConnectivityExceptions, we will continue to recalculate the number of maxRetryAttempts - so this maxRetryAttempts could potentially change during every loop?  is this expected/intended?  (the behavior didnt change in this diff, but just something to be aware of)

Reply:
The retryAttempts are calculated for the first ServerConnectivityException; which changes the init value of maxRetryAttempts; in the following code, we check to see if its less than 0 and return from the call (breaking from the for loop).